vms/platformvm/spend.go

```diff
@@ -357,14 +359,15 @@ func (vm *VM) semanticVerifySpend(
 		utxos[index] = utxo
 	}
 
-	return vm.semanticVerifySpendUTXOs(tx, utxos, ins, outs, creds, feeAmount, feeAssetID)
+	return vm.semanticVerifySpendUTXOs(tx, utxos, ins, outs, creds, feeAmount, feeAssetID, credsOffset)
 }
 
 // Verify that [tx] is semantically valid.
 // [db] should not be committed if an error is returned
 // [ins] and [outs] are the inputs and outputs of [tx].
 // [creds] are the credentials of [tx], which allow [ins] to be spent.
 // [utxos[i]] is the UTXO being consumed by [ins[i]]
+// [credsOffset] Length of [ins] should be equal length[creds] - [credsOffset]. The remaining creds will be ignored
 // Precondition: [tx] has already been syntactically verified
 func (vm *VM) semanticVerifySpendUTXOs(
 	tx UnsignedTx,
```

Comment:
Here I think it will be better to just remove the offending cred from the array prior to the function call via a slice operation. Add a comment why it's done and then I think it's a better solution that to change a core functions, meaning keeping the required hacks localized.

Reply:
good point!